File: backend/src/InsuraTech.API/Controllers/VehiclePlansController.cs
using InsuraTech.Application.VehiclePlans.DTOs;
using InsuraTech.Application.VehiclePlans.Queries.CalculateVehiclePlans;
using InsuraTech.Application.VehiclePlans.Queries.GetVehiclePlans;
using MediatR;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace InsuraTech.API.Controllers;

[ApiController]
[Authorize]
[Route("api/v1/vehicle-plans")]
[Produces("application/json")]
public sealed class VehiclePlansController : ControllerBase
{
    private readonly IMediator _mediator;

    public VehiclePlansController(IMediator mediator) => _mediator = mediator;

    /// <summary>Lista los tres planes de cobertura vehicular con sus multiplicadores y asistencias.</summary>
    [HttpGet]
    [ProducesResponseType(typeof(IReadOnlyList<VehiclePlanDto>), StatusCodes.Status200OK)]
    public async Task<IActionResult> GetAll(CancellationToken cancellationToken)
    {
        var result = await _mediator.Send(new GetVehiclePlansQuery(), cancellationToken);
        return Ok(result);
    }

    /// <summary>
    /// Ejecuta el motor de cotización vehicular y retorna las primas de los tres planes.
    /// </summary>
    /// <param name="commercialValue">Valor comercial del vehículo (COP). Debe ser mayor a 0.</param>
    /// <param name="vehicleYear">Año de fabricación del vehículo.</param>
    /// <param name="brand">Marca del vehículo.</param>
    [HttpGet("calculate")]
    [ProducesResponseType(typeof(VehicleQuotationDto), StatusCodes.Status200OK)]
    [ProducesResponseType(StatusCodes.Status400BadRequest)]
    [ProducesResponseType(StatusCodes.Status422UnprocessableEntity)]
    public async Task<IActionResult> Calculate(
        [FromQuery] decimal commercialValue,
        [FromQuery] int     vehicleYear,
        [FromQuery] string  brand,
        CancellationToken cancellationToken)
    {
        var result = await _mediator.Send(
            new CalculateVehiclePlansQuery
            {
                CommercialValue = commercialValue,
                VehicleYear     = vehicleYear,
                Brand           = brand
            },
            cancellationToken);

        return Ok(result);
    }
}

File: backend_v2/src/main/java/com/insuratech/api/controllers/VehiclePlansController.java
// Spec: SPEC-010 — planes vehiculares
package com.insuratech.api.controllers;

import com.insuratech.application.vehicleplans.VehiclePlanService;
import com.insuratech.application.vehicleplans.dto.VehiclePlanDto;
import com.insuratech.application.vehicleplans.dto.VehicleQuotationDto;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.math.BigDecimal;
import java.util.List;

@RestController
@RequestMapping("/v1/vehicle-plans")
@RequiredArgsConstructor
public class VehiclePlansController {

    private final VehiclePlanService vehiclePlanService;

    @GetMapping
    public ResponseEntity<List<VehiclePlanDto>> getAll() {
        return ResponseEntity.ok(vehiclePlanService.getAll());
    }

    @GetMapping("/calculate")
    public ResponseEntity<VehicleQuotationDto> calculate(
            @RequestParam BigDecimal commercialValue,
            @RequestParam int vehicleYear,
            @RequestParam String brand) {
        return ResponseEntity.ok(vehiclePlanService.calculate(commercialValue, vehicleYear, brand));
    }
}
